app/controllers/internal/reactions_controller.rb

```diff
@@ -5,9 +5,12 @@ class Internal::ReactionsController < Internal::ApplicationController
 
   def update
     @reaction = Reaction.find(params[:id])
-    @reaction.update(status: params[:reaction][:status])
-    Moderator::SinkArticles.call(@reaction.reactable_id) if confirmed_vomit_reaction?
-    redirect_to "/internal/reports"
+    if @reaction.update(status: params[:status])
+      Moderator::SinkArticles.call(@reaction.reactable_id) if confirmed_vomit_reaction?
+      render json: { outcome: "Success" }
+    else
+      render json: { outcome: "Failed to update reaction" }
```

Comment:
this could be

```ruby
render json: { error: reaction.errors.full_messages.to_sentence }, status: :unprocessable_entity
```

What would you gain? Two things that are helpful in the frontend:

1. a clear HTTP error code, `:unprocessable_entity` will turn into [HTTP 422](https://httpstatuses.com/422) for the JS client
1. the error message, which could be ignored or displayed, depends on how far you want to go

Reply:
great point!